core/src/main/web/app/mainapp/components/pluginmanager/pluginmanager-directive.js

```diff
@@ -145,5 +145,19 @@
       }
     };
 
+    $scope.$on(GLOBALS.EVENTS.LANGUAGE_MANAGER_SHOW_SPINNER, function(event, data) {
+      $scope.loading = true;
+      $scope.loadingMessage = 'Restarting ' + data.pluginName + '...';
+      if (!($scope.$$phase || $rootScope.$$phase))
+        $scope.$digest();
```

Comment:
$on handler would have to be in the digest loop already, no? Are you sure we need to kick off the digest loop here?